source/common/http/conn_manager_impl.h

```diff
@@ -279,6 +279,8 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     void decodeHeaders(ActiveStreamDecoderFilter* filter, HeaderMap& headers, bool end_stream);
     void decodeData(ActiveStreamDecoderFilter* filter, Buffer::Instance& data, bool end_stream);
     void decodeTrailers(ActiveStreamDecoderFilter* filter, HeaderMap& trailers);
+    void disarmRequestTimer();
+    void maybeDisarmRequestTimer();
```

Comment:
Sorry, should have specified, but I'd like it in the header file.  IMO, it does no good in the .cc file because the reader can see exactly what it's saying three lines later in the code.  My goal is for header files to capture the interface contract for routines--i.e. code that's trying to use the interface should (in theory, and yes, theory and practice :-J) be able to figure out how and what to call from looking at the header file.  

(I think of people as looking first in header files for clues as to how to use interfaces or what the intent behind the code was.)

Having said that, I'll downgrade this to optional in that I think the documentation around the timeout is pretty clear.  But do remove the comment in the .cc file; that's just clutter.